src/Microsoft.Azure.WebJobs.Host/Bindings/ConverterManager.cs

```diff
@@ -226,6 +226,11 @@ private static OpenType GetOpenType(Type t)
                 }
             }
 
+            if (typeDest == typeof(JArray) && typeof(IEnumerable).IsAssignableFrom(typeSource))
```

Comment:
T->S and its usage to convert enumerable types makes sense, but I agree that there's a lot of complexity there.

I'm proposing that we support Type -> JObject and IEnumerable<Type> -> JArray out of box with the FromObject methods. It will light up the script functionality out of the box without the extension author needing to worry about it, and if it doesn't work for their particular scenario they can define converters to override the behavior. What drawbacks do you see to this?



---
In reply to: [102851157](https://github.com/Azure/azure-webjobs-sdk/pull/1037#discussion_r102851157) [](ancestors = 102851157,102849948)